app/components/onboarding_channel_buttons/onboarding_channel_buttons.rb

```diff
@@ -4,14 +4,19 @@ module OnboardingChannelButtons
   class OnboardingChannelButtons < ApplicationComponent
     private
 
-    def styles
-      return super unless show_signal?
+    def initialize(channels:)
+      super
 
-      super + [:even]
+      @channels = channels
     end
 
-    def show_signal?
-      Setting.signal_server_phone_number.present?
+    attr_reader :channels
+
+    def styles
+      return super if (channels.length % 3).zero?
+      return super + [:twoColumn] if channels.length.even?
```

Comment:
Ok, I've simplified it in f33783fe
It's not perfect since the styling assumes up to 6 channels being configured. If we end up adding 7 channels, it won't look broken, but we might want to reconsider the styling.